api/proto/metrics/metrics.proto

```diff
@@ -468,6 +468,43 @@ message ZMetricVolume {
   uint64 freeBytes = 9;
 }
 
+// Metrics for a snapshot
+// When a snapshot is created, its disk space is initially shared between
+// the snapshot and the file system, and possibly with previous snapshots.
+// As the file system changes, disk space that was previously shared becomes
+// unique to the snapshot, and thus is counted in the snapshot's used property.
+// Additionally, deleting snapshots can increase the amount of disk space
+// unique to (and thus used by) other snapshots, however, this does not increase
+// the amount of disk space specified for the volume itself.
+message ZMetricSnapshot {
```

Comment:
Can we also send `written` field as part of `ZMetricSnapshot` message? This field indicates the difference between snapshots and will help us to see references unique for the snapshot against the previous one.